### DISCUSSION TITLE
Bosean devices: Firmware update over USB

### DISCUSSION THREAD

**Discussion by JTknives** [Gissio/radpro#102] [Q&A]
I am working on a Geiger Counter for resale that's based on the fs-5000. I am wondering if there is a way to update the firmware over usb after it's been flashed. I don't want to require customers to open up the case to get access to an installed plug and use a supplied st-link. I am working on a way to build the st-Link into the case and just leave it there. But it would be much nicer to just be-able to update over the built in USB C port. Thanks

**Comment by Gissio** [comment]
Please go to the "Security" tab, click on "Report a vulnerability" and post your email address so I can contact you. Your email address won't be published.

**Reply by JTknives** [reply]
Ok, did that. Thank you for your help. This was my current attempt at tying the st-link directly to the boards USB but I could not get it to work. It got power but would not flash.

 ![image](https://github.com/Gissio/radpro/assets/171413119/83c9f55a-32b0-4fd9-adff-08448bea0f10)
